libraries/standard/mqtt/src/mqtt.c

```diff
@@ -969,9 +1000,12 @@ static MQTTStatus_t receiveConnack( const MQTTContext_t * pContext,
         }
 
         /* Reading the remainder of the packet by transport recv.
-         * Attempt to read once even if the timeout has expired at this point.
+         * Attempt to read once even if the timeout has expired.
          * Invoking receivePacket with remainingTime as 0 would attempt to
-         * recv from network once.*/
+         * recv from network once. If using retries, the remainder of the
```

Comment:
We do not re-try even if retries are left?

Reply:
No. We don't. After reading the packet type and length, it is the remaining 2 bytes that we would need to read. The delay is required to wait for the reception of CONNACK packet after sending a CONNECT packet. Once CONNACK packet type and length is available, we would need a follow up read just to read the remaining 2 bytes.  Also in this case, we would be expecting only one packet from broker.